idaes/conftest.py

```diff
@@ -29,37 +29,72 @@
-#
-# import pytest
-# @pytest.mark.linux
-# def test_something():
-#    print("this only runs on linux")
-#
-#
-# In addition, you can use "no<platform>" to exclude
-#
-# @pytest.mark.nowin32
-# def test_something():
-#    print("this will not run on windows")
-#
-# The names of the platforms should match what is returned by `sys.platform`, in particular:
-#    Linux = 'linux'
-#    Windows = 'win32'
-#    macOS = 'darwin'
 
 
-ALL = {"darwin", "linux", "win32"}
-ALL_NO = {"no" + tag for tag in ALL}
+@pytest.hookimpl
+def pytest_runtest_setup(item):
+    _validate_required_markers(
+        item,
+        required_markers={"unit", "component", "integration"},
+        expected_count=1
+    )
+    _skip_for_unsupported_platforms(
+        item,
+        all_platforms={"darwin", "linux", "win32"},
+        negate_tag=(lambda tag: f'no{tag}'),
+    )
+
 
+def _skip_for_unsupported_platforms(item, all_platforms=None, negate_tag=None):
+    """
+    Only run the tests for your particular platform(s), if it is marked with those platform(s)
+    e.g. to mark tests for linux:
 
-def pytest_runtest_setup(item):
-    supported_platforms = ALL.intersection(mark.name for mark in item.iter_markers())
-    excluded_platforms = ALL_NO.intersection(mark.name for mark in item.iter_markers())
+    import pytest
+    @pytest.mark.linux
+    def test_something():
+       print("this only runs on linux")
+
+    In addition, you can use "no<platform>" to exclude
+
+    @pytest.mark.nowin32
+    def test_something():
+       print("this will not run on windows")
+
+    The names of the platforms should match what is returned by `sys.platform`, in particular:
+       Linux = 'linux'
+       Windows = 'win32'
+       macOS = 'darwin'
+    """
+
+    all_platforms = set(all_platforms or [])
+    if negate_tag is None:
+        def negate_tag(tag):
+            return f'no{tag}'
+    all_negated_platforms = {negate_tag(tag) for tag in all_platforms}
+    item_markers = {marker.name for marker in item.iter_markers()}
+    supported_platforms = all_platforms & item_markers
+    excluded_platforms = all_negated_platforms & item_markers
     plat = sys.platform
-    if ((excluded_platforms and ("no" + plat) in excluded_platforms) or
+    if ((excluded_platforms and negate_tag(plat) in excluded_platforms) or
         (supported_platforms and plat not in supported_platforms)):
         pytest.skip("cannot run on platform {}".format(plat))
 
+
+def _validate_required_markers(item, required_markers=None, expected_count=1):
+    required_markers = set(required_markers or [])
+    item_markers = {marker.name for marker in item.iter_markers()}
+    required_markers_on_item = item_markers & required_markers
+    required_count = len(required_markers_on_item)
+    reason_to_fail = None
+    if required_count < expected_count:
+        reason_to_fail = 'Too few required markers'
+    if required_count > expected_count:
+        reason = 'Too many required markers'
```

Comment:
I think this should be `reason_to_fail`, as well

Reply:
Good catch, thanks! I'll change it to `reason_to_fail`.